src/ios-deploy/ios-deploy.m

```diff
@@ -987,6 +989,7 @@ void fdvendor_callback(CFSocketRef s, CFSocketCallBackType callbackType, CFDataR
     lldb_socket  = CFSocketCreateWithNative(NULL, socket, kCFSocketDataCallBack, &lldb_callback, NULL);
     int flag = 1;
     int res = setsockopt(socket, IPPROTO_TCP, TCP_NODELAY, (char *) &flag, sizeof(flag));
+    (void)res; // Disable unused variable warning.
```

Comment:
I think we should probably refactor the asserts rather than disabling the warning.

This assert is doing what check_error() does. We should probably just rewrite this to be 
```
check_error(setsockopt(socket, IPPROTO_TCP, TCP_NODELAY, (char *) &flag, sizeof(flag)));
```
** I didn't actually try that code **

Reply:
check_error is specifically for checking for MobileDevice errors. Best case our error code doesn't match any MobileDevice errors and we get "unknown" error message, worst we get a false positive and print something unrelated.

Reply:
ah, true about the error printing, but I'm for either removing the variable and not checking at all or checking and printing an error.

Reply:
SGTM, there was another on_sys_error() function that works with errno and fread, getsockname, and setsockopt all set errno so switched asserts to on_sys_error